src/cached_object_store/lru.rs

```diff
@@ -0,0 +1,502 @@
+#![allow(clippy::needless_lifetimes)]
+#![allow(clippy::extra_unused_lifetimes)]
+#![allow(clippy::unwrap_used)]
+#![allow(unused)]
+//! An implementation of a generic LRU cache. It supports `get`, `put`, `remove`,
```

Comment:
Given that this is considered a generic LRU cache implementation, what are your thoughts on choosing an existing crate for this?"

"I'm concerned that implementing a robust LRU cache might involve some complexities, and these complexities could be better handled by using an existing crate